### PR TITLE
feat: store chain of thought in DB

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- adds a `chainOfThoughts` string array column on `agentMessage`
- properly save the CoT tokens in that new column
- add an event for CoT (we emit it if there is any CoT just before we emit agent actions event. We also emit if there is any content at this point, because that content becomes a posteriori CoT)

## Risk

critical path + db migration

## Deploy Plan

apply migration (prodbox)
deploy code

### CODE REVIEW

File: front/lib/api/assistant/conversation.ts [hunk 1]
```diff
@@ -1712,9 +1718,18 @@ async function* streamRunAgentEvents(
 
       case "agent_generation_success":
         // Store message in database.
-        await agentMessageRow.update({
+        const updateParams: Parameters<typeof agentMessageRow.update>[0] = {
           content: event.text,
-        });
+        };
+
+        if (event.chainOfThought.length) {
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/conversation.ts:1725]: Yeah fair, I hesitated but let's make it a CoT event.

File: front/lib/api/assistant/agent.ts [hunk 2]
```diff
@@ -722,6 +732,25 @@ export async function* runMultiActionsAgent(
...
+    yield {
+      type: "agent_chain_of_thought",
+      created: Date.now(),
+      configurationId: agentConfiguration.sId,
+      messageId: agentMessage.sId,
+      message: agentMessage,
+      // In practice, we should never have both chainOfThought and content.
+      // It is not completely impossible that eg Anthropic decides to emit part of the
+      // CoT between `<thinking>` XML tags and the rest outside of any tag.
+      chainOfThought: removeNulls([chainOfThought, content]).join("\n"),
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/api/assistant/agent.ts:750]: Let's add a comment to make it clearer along the lines of:

All content here was generated before a tool use and is not proper generation content and can therefore be safely assumed to be reflection from the model before using a tool.

[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/agent.ts:762]: Done

File: types/src/front/lib/api/assistant/agent.ts [hunk 3]
```diff
@@ -81,6 +81,7 @@ export type AgentGenerationSuccessEvent = {
   configurationId: string;
   messageId: string;
   text: string;
+  chainOfThought: string;
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply types/src/front/lib/api/assistant/agent.ts]: Fixed

**Review by spolu — APPROVED** [review]

**Pull request merged**